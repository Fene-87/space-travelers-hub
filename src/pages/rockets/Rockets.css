.rockets-container {
  padding-bottom: 5rem;
}
