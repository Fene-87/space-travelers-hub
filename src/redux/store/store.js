import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from '../rockets/rocketsSlice';
import missionReducer from '../missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    missions: missionReducer,
  },
});

export default store;
