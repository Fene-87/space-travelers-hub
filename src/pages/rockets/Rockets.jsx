import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Navbar from '../../components/navbar/Navbar';
import Rocket from '../../components/rocket/Rocket';
import { fetchRockets } from '../../redux/rockets/rocketsSlice';
import './Rockets.css';

const Rockets = () => {
  const { rocketList, status } = useSelector((store) => store.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchRockets());
    }
  }, [status, dispatch]);

  return (
    <div>
      <Navbar />
      <section className="rockets-container">
        {rocketList.map((item) => (
          <Rocket key={item.id} {...item} />
        ))}
      </section>
    </div>
  );
};

export default Rockets;
