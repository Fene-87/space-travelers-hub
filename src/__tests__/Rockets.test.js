import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store/store';
import Rockets from '../pages/rockets/Rockets';

describe('testing snapshot with Rockets', () => {
    it('testing Rockets component', () => {
        const rockets = render(
        <Provider store={store}>
            <BrowserRouter>
            <Rockets />
            </BrowserRouter>
        </Provider>,
        );
        expect(rockets).toMatchSnapshot();
    });
});