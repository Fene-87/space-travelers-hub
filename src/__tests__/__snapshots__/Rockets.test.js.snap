// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshot with Rockets testing Rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="navbar"
        >
          <div
            class="left-side"
          >
            <img
              alt="logo"
              class="logo-img"
              src="planet.png"
            />
            <h2
              class="nav-header"
            >
              Space Travelers' Hub
            </h2>
          </div>
          <div
            class="right-side"
          >
            <ul
              class="nav-links"
            >
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  Rockets
                </a>
              </li>
              <li
                class="boundary"
              >
                <a
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/profile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </section>
        <section
          class="rockets-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="navbar"
      >
        <div
          class="left-side"
        >
          <img
            alt="logo"
            class="logo-img"
            src="planet.png"
          />
          <h2
            class="nav-header"
          >
            Space Travelers' Hub
          </h2>
        </div>
        <div
          class="right-side"
        >
          <ul
            class="nav-links"
          >
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li
              class="boundary"
            >
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class=""
                href="/profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </section>
      <section
        class="rockets-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
